### PR TITLE
MINOR FIX: Enable Optional Stringification of Properties as StringCon…

### LINKED ISSUE #130: Allow or skip nullable values to be part of the multipart form
Describe issue:

Currently, we cannot construct a MultipartForm for a model which has null or empty values.

Given the model:

```csharp
public class Person
{
    [RESTFulStringContent(name: "firstName")]
    public string FirstName { get; set; }

    [RESTFulStringContent(name: "middleName")]
    public string? MiddleName { get; set; }
}

async Task MakeRequestAsync()
{
    var person = new Person
    {
        FirstName = "John Smith",
        MiddleName = null
    };

    var httpClient = SetupHttpClient();
    var apiClient = new RESTFulApiFactoryClient(httpClient);

    var response = await apiClient.PostFormAsync<Person, string>("/users", person);
    // Since no value was provided for middleName property, next exception is thrown:
    //
    // RESTFulSense.Models.Coordinations.Forms.Exceptions.FormCoordinationDependencyValidationException : Form coordination dependency validation error occurred, fix the errors and try again.
    // ---- RESTFulSense.Models.Orchestrations.Forms.Exceptions.FormOrchestrationDependencyValidationException : Form orchestration dependency validation error occurred, fix the errors and try again.
    // -------- RESTFulSense.Models.Foundations.Forms.Exceptions.FormValidationException : Form validation error occurred, fix errors and try again.
    // ------------ RESTFulSense.Models.Foundations.Forms.Exceptions.InvalidFormArgumentException : Invalid form arguments.Please fix the errors and try again.
    // Stack Trace:
    // at RESTFulSense.Services.Coordinations.Forms.FormCoordinationService.TryCatch(ReturningMultipartFormDataContentFunction returningMultipartFormDataContentFunction)
    // at RESTFulSense.Services.Coordinations.Forms.FormCoordinationService.ConvertToMultipartFormDataContent[T](T object)
    // at RESTFulSense.Clients.RESTFulApiFactoryClient.PostFormAsync[TContent, TResult](String relativeUrl, TContent content, CancellationToken cancellationToken)
    // ...

}
```

### LINKED ISSUE #132: Cannot create a multipart-form with other types except string
When trying to construct a multipart-form for a model which has properties of another type except for string type, the next exception is thrown.

```csharp
public class Person
{
    [RESTFulStringContent(name: "firstName")]
    public string FirstName { get; set; }

    [RESTFulStringContent(name: "sallary")]
    public decimal Salary { get; set; }
}

async Task MakeRequestAsync()
{
    var person = new Person
    {
        FirstName = "Jane Smith",
        Salary = 10_500M
    };

    var httpClient = SetupHttpClient();
    var apiClient = new RESTFulApiFactoryClient(httpClient);

    var response = await apiClient.PostFormAsync<Person, string>("/users", person);
}
```

Exception thrown:

```
RESTFulSense.Models.Client.Exceptions.RESTFulApiClientDependencyException : Form coordination dependency error occurred, fix the errors and try again.
---- RESTFulSense.Models.Orchestrations.Forms.Exceptions.FormOrchestrationServiceException : Form orchestration service error occurred, contact support.
-------- RESTFulSense.Models.Orchestrations.Forms.Exceptions.FailedFormOrchestrationServiceException : Failed form orchestration service occurred, please contact support
------------ System.InvalidCastException : Unable to cast object of type 'System.Decimal' to type 'System.String'.
```

### PR DESCRIPTION
**Pull request by BrianLParker**
@hassanhabib @Catalin-Andronie

Closes #130
Closes #132

[POC Repo](https://github.com/BrianLParker/TranscriptionPostPoc) (Reference this branch ...)

Output Screenshot
image

### CODE REVIEW

File: RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs [hunk 1]
```diff
@@ -74,12 +76,62 @@ private void AddStringContents(FormModel formModel)
...
+            }
+        }
+
+        private static string ConvertToString(object propertyValue, string mediaType, bool ignoreDefaultValues)
+        {
+            return mediaType switch
+            {
+                "text/json" => ConvertToJsonString(propertyValue, ignoreDefaultValues),
+                "application/json" => ConvertToJsonString(propertyValue, ignoreDefaultValues),
+                _ => propertyValue.ToString()
```
Comment (Catalin-Andronie) [inline RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs:112]: We should guard for null `propertyValue`. This code path is triggered when `property` has a default value of null and we don't ignore default values.

```suggestion
                _ => propertyValue?.ToString()
```

File: RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs [hunk 2]
```diff
@@ -74,12 +76,62 @@ private void AddStringContents(FormModel formModel)
...
+                    }
                 }
             }
         }
 
+        private static bool IsDefaultValue(Type type, object value)
+        {
+            if (type.IsValueType)
+            {
+                object defaultValue = Activator.CreateInstance(type);
```
Comment (Catalin-Andronie) [inline RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs:96]: Value types may be null, if that happens when checking equality we'll encounter a null exception. We should verify if the underlying type is Nullabe<T> and if it is then its default value is null as well.

```suggestion
                Type underlyingType = Nullable.GetUnderlyingType(type);
                bool isNullableType = underlyingType != null;
                if (isNullableType && value is null)
                {
                    return true;
                }

                object defaultValue = Activator.CreateInstance(type);
```

**Review by Catalin-Andronie — COMMENTED** [review]
@BrianLParker please check my remarks.

**Comment by hassanhabib** [comment]
@BrianLParker @Catalin-Andronie I will review this well.

File: RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs [hunk 3]
```diff
@@ -109,7 +109,7 @@ private static string ConvertToString(object propertyValue, string mediaType, bo
             {
                 "text/json" => ConvertToJsonString(propertyValue, ignoreDefaultValues),
                 "application/json" => ConvertToJsonString(propertyValue, ignoreDefaultValues),
-                _ => propertyValue.ToString()
+                _ => propertyValue?.ToString()
```
Comment (hassanhabib) [inline RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs:112]: Where's the test for this Null Guard?

Comment (BrianLParker) [reply RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs:112]: This is still draft the tests are coming. I need to talk re tests and optional serialization as well.

**Comment by Catalin-Andronie** [comment]
@hassanhabib @BrianLParker any news about this?
We need this to be merged in order to continue implementing [CLIENTS: Transcript Audio File](https://github.com/hassanhabib/Standard.AI.OpenAI/issues/212).

**Pull request closed without merging**